rust/operator-binary/src/crd.rs

```diff
@@ -175,9 +175,35 @@ pub enum KerberosKeytabBackendAdmin {
         /// The root Distinguished Name (DN) for AD-managed schemas,
         /// typically `CN=Schema,CN=Configuration,{domain_dn}`.
         schema_distinguished_name: String,
+
+        /// Allows samAccountName generation for new accounts to be customized.
+        /// Note that setting this field (even if empty) makes the Secret Operator take
+        /// over the generation duty from the domain controller.
+        experimental_generate_sam_account_name: Option<ActiveDirectorySamAccountNameRules>,
     },
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct ActiveDirectorySamAccountNameRules {
+    /// A prefix to be prepended to generated samAccountNames.
+    #[serde(default)]
+    pub prefix: String,
+    /// The total length of generated samAccountNames, _including_ `prefix`.
+    /// Must be larger than the length of `prefix`, but at most `20`.
```

Comment:
I could not find any check for a max length of 20, should we add a snafu error for this?

Reply:
It is the limit documented by Microsoft.

I don't think it's worth us validating it, if AD is happy with a longer value then we don't particularly care.

Reply:
Works for me